content/blog/kubernetes-monitoring-using-prometheus-and-grafana-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,419 @@
+
+In the following sections, I will show you how to add a monitoring stack using Prometheus and Grafana to a K8s cluster deployed in HPE GreenLake for Private Cloud Enterprise.
+
+### Deploy Prometheus and Grafana using Terraform
+
+Prometheus and Gafana will be deployed to the K8s cluster using the [HPE GreenLake Terraform provider *hpegl*](https://registry.terraform.io/providers/HPE/hpegl/latest), together with the [Helm provider from Hashicorp](https://registry.terraform.io/providers/hashicorp/helm/latest). 
+
+#### Create a Terraform config
+
+H﻿ere is the terraform config file. You can refer to [Infrastructure-as-code on HPE GreenLake using Terraform](https://developer.hpe.com/blog/infrastructure-as-code-on-hpe-greenlake-using-terraform/) for the details about HPE GreenLake Terraform provider and its usage.
+
+```markdown
+$ cat main.tf 
+terraform {
+  required_providers {
+    hpegl = {
+      source  = "hpe/hpegl"
+      version = ">= 0.2.2"
+    }
+  }
+}
+
+provider "hpegl" {
+  caas {
+    api_url = var.api_url
+  }
+}
+
+data "hpegl_caas_cluster" "iaccluster" {
+  name     = var.cluster_name
+  space_id = var.space_id
+}
+
+provider "helm" {
+  kubernetes {
+       host     = yamldecode(base64decode(data.hpegl_caas_cluster.iaccluster.kubeconfig)).clusters[0].cluster.server
+       token    = yamldecode(base64decode(data.hpegl_caas_cluster.iaccluster.kubeconfig)).users[0].user.token
+       insecure = true
+  }
+}
+
+resource "helm_release" "prometheus-stack" {
+   name = "prometheus-stack"
+   repository = "https://prometheus-community.github.io/helm-charts"
+   chart = "prometheus"
+   version = "23.0.0"
+   namespace = "monitoring"
+   create_namespace = true
+
+   set {
+    name  = "server.service.type"
+    value = "NodePort"
+  }
+
+   set {
+    name  = "prometheus-node-exporter.hostRootFsMount.enabled"
+    value = "false"
+  }
+
+   set {
+    name  = "prometheus-node-exporter.hostNetwork"
+    value = "false"
+  }
+
+   set {
+    name  = "prometheus-node-exporter.hostPID"
+    value = "false"
+  }
+}
+
+resource "helm_release" "grafana-dashboard" {
+   name = "grafana-dashboard"
+   repository = "https://grafana.github.io/helm-charts"
+   chart = "grafana"
+   version = "6.57.4"
+   namespace = "monitoring"
+   create_namespace = true
+
+   set {
+    name  = "service.type"
+    value = "NodePort"
+  }
+
+   set {
+    name  = "persistence.enabled"
+    value = "true"
+  }
+}
+```
+
+There are a few things worth noting in above Terraform config file:
```

Comment:
Modify to:
"There are a few things worth noting in **the** above Terraform **configuration** file."**** (added the word "the" and spelled out "configuration")